router/customdestinationmanager/customdestinationmanager_test.go

```diff
@@ -0,0 +1,100 @@
+package customdestinationmanager
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/config"
+	backendconfig "github.com/rudderlabs/rudder-server/config/backend-config"
+	"github.com/rudderlabs/rudder-server/services/stats"
+	"github.com/rudderlabs/rudder-server/services/streammanager/kafka"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCircuitBreaker(t *testing.T) {
+	const (
+		normalError  = "could not ping: could not dial tcp/unknown.example.com:9999: failed to dial: failed to open connection to unknown.example.com:9999: dial tcp: lookup unknown.example.com: no such host"
+		breakerError = "circuit breaker is open"
+	)
+
+	// init various packages
+	config.Load()
+	Init()
+	logger.Init()
+	stats.Setup()
+	kafka.Init()
+
+	// don't let manager subscribe to backend-config
+	skipBackendConfigSubscriber = true
+
+	// set a custom circuit breaker timeout (default is 60 seconds)
+	breakerTimeout := 1 * time.Second
+
+	manager := New("KAFKA", Opts{Timeout: 1 * time.Microsecond}).(*CustomManagerT)
+	manager.breakerTimeout = breakerTimeout
+
+	dest := getDestConfig()
+	// during the first 6 failed attempts the circuit is closed
+	count := 1
+	newDestination(t, manager, dest, count, normalError)
+	count++
+	for ; count <= 6; count++ {
+		newClientAttempt(t, manager, dest.ID, count, normalError)
+	}
+	// after the 6th failed attempt the circuit opens
+	newClientAttempt(t, manager, dest.ID, count, breakerError)
+	count++
+	time.Sleep(breakerTimeout)
```

Comment:
It would be preferable if we use a channel to control the time instead of sleep here